### PR TITLE
Fix CustomViews by switching to WebViews

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

I'm going to try to see if I can add a test or two... but this fixes the *-VSCodeHtmlContentView cmdlets. You are still able to pass in stylesheets and javascript and it will inline them.

Then, in order to refresh your view to get any js or css updates, just run the `Show-VSCodeHtmlContentView` cmdlet again and your changes will be reflected.

This also enables the persistent pages so if you click away from a view and come back, the state remains!

P.S. there seems to be some weird behavior when you close out of a view - we must not be cleaning up properly. This bug existed before this change... I might try to tackle it now.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Ok rewire is back but with it, comes 3 tests!

**Comment by TylerLeonhardt** [comment]
Implementation has changed a bit - re-requested reviews

File: src/features/CustomViews.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
 
+import path = require("path");
```
Comment (rjmholt) [inline src/features/CustomViews.ts]:
```suggestion
import path from 'path';
```

File: src/features/CustomViews.ts [hunk 2]
```diff
@@ -2,8 +2,9 @@
...
 import vscode = require("vscode");
```
Comment (rjmholt) [inline src/features/CustomViews.ts]:
```suggestion
import vscode from 'vscode';
```

File: src/features/CustomViews.ts [hunk 3]
```diff
@@ -179,42 +176,57 @@ class HtmlContentView extends CustomView {
...
-
-        function getNonce(): number {
-            return Math.floor(Math.random() * 100000) + 100000;
-        }
-
         if (this.htmlContent.styleSheetPaths &&
             this.htmlContent.styleSheetPaths.length > 0) {
-            styleSrc = "";
             this.htmlContent.styleSheetPaths.forEach(
                 (p) => {
```
Comment (rjmholt) [inline src/features/CustomViews.ts]: Possible to change `p` to be more descriptive (to `path` maybe)? Is that the path to a css file?

File: test/features/CustomViews.test.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import * as assert from "assert";
+import fs = require("fs");
+import path = require("path");
+import rewire = require("rewire");
+import vscode = require("vscode");
+
+// Setup types that are not exported.
+const customViews = rewire("../../src/features/CustomViews");
+const htmlContentViewClass = customViews.__get__("HtmlContentView");
+const HtmlContentView: typeof htmlContentViewClass = htmlContentViewClass;
```
Comment (rjmholt) [inline test/features/CustomViews.test.ts]: Wow, this is cool!

Comment (TylerLeonhardt) [reply test/features/CustomViews.test.ts]: It's like reflection but much more convoluted lol

File: src/features/CustomViews.ts (hunk 3 above)
Comment (TylerLeonhardt) [reply src/features/CustomViews.ts]: made it p so it didn't conflict with `import path from 'path';`... I'll change it to `styleSheetPath` or something

File: src/features/CustomViews.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/features/CustomViews.ts]: `import * as vscode from 'vscode'` (for clarity)

**Pull request merged**